node/src/effect.rs

```diff
@@ -1536,23 +1536,23 @@ impl<REv> EffectBuilder<REv> {
     }
 
     /// Gets the correct era validators set for the given era.
-    /// Takes upgrades and emergency restarts into account based on the `initial_state_root_hash`
-    /// and `activation_era_id` parameters.
+    /// Takes emergency restarts into account based on the information from the chainspec loader.
     pub(crate) async fn get_era_validators(self, era_id: EraId) -> Option<BTreeMap<PublicKey, U512>>
     where
         REv: From<ContractRuntimeRequest> + From<StorageRequest> + From<ChainspecLoaderRequest>,
     {
         let CurrentRunInfo {
-            activation_point,
             protocol_version,
             initial_state_root_hash,
+            last_emergency_restart,
+            ..
         } = self.get_current_run_info().await;
-        let activation_era_id = activation_point.era_id();
-        if era_id < activation_era_id {
-            // we don't support getting the validators from before the last upgrade
+        let cutoff_era_id = last_emergency_restart.unwrap_or_else(|| EraId::new(0));
```

Comment:
So this can be earlier than the previous activation point, right?
Is that safe? What if there was an upgrade _after_ the last emergency restart which changed the global state format, or the auction contract state's format?

Reply:
I think it should be safe. Even if the format changed, the global state should be migrated and we're reading it through contract runtime, which should be able to deal with any changes. Or am I wrong?

Reply:
I agree this should be safe too.